cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1243,6 +1243,24 @@
              (x64_psrlq (vector_all_ones $F64X2)
                     (RegMemImm.Imm 1))))
 
+;;;; Rules for `fneg` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type $F32 (fneg x)))
+      (x64_xorps x (imm $F32 0x80000000)))
+
+(rule (lower (has_type $F64 (fneg x)))
+      (x64_xorpd x (imm $F64 0x8000000000000000)))
+
+(rule (lower (has_type $F32X4 (fneg x)))
+      (x64_xorps x
+             (x64_pslld (vector_all_ones $F32X4)
```

Comment:
I wonder, would it be better to use a VCodeConstant here to get the appropriate MSBs-only-in-each-lane value?

Reply:
Would that be a savings over the two instructions?

Reply:
Possibly, though looking at `xorps` I see the basic SSE version is not `reg, reg/mem` but just `reg, reg` so it would still be a separate load; one load vs. dependent chain of two 1-cycle ops, the latter likely wins. Anyway it's an unimportant enough question that I don't care too much to investigate further :-)